tests/integ/modin/io/test_to_view.py

```diff
@@ -0,0 +1,406 @@
+        self,
+        session,
+        db_parameters,
+    ) -> None:
+        try:
+            # create table
+            table_name = Utils.random_table_name()
+            session.create_dataframe(
+                [BASIC_TYPE_DATA1, BASIC_TYPE_DATA2]
+            ).write.save_as_table(table_name)
+
+            # create dataframe with enforce_ordering enabled
+            snow_dataframe = pd.read_snowflake(
+                f"(((SELECT * FROM {table_name})))", enforce_ordering=True
+            )
+
+            # create view
+            view_name = Utils.random_view_name()
+            assert (
+                "successfully created"
+                in pd.to_view(obj=snow_dataframe, name=view_name)[0]["status"]
+            )
+
+            # another session
+            new_session = Session.builder.configs(db_parameters).create()
+            pd.session = session
+
+            # accessing the created view in another session fails when enforce_ordering is enabled
+            with pytest.raises(
+                SnowparkSQLException,
+                match="Object 'VIEW_NAME' does not exist or not authorized",
+            ):
+                new_session.sql("select * from view_name").collect()
+            new_session.close()
+
+        finally:
+            # cleanup
+            Utils.drop_view(session, view_name)
+            Utils.drop_table(session, table_name)
+            pd.session = session
+
+    @sql_count_checker(query_count=5)
+    def test_to_view_multiple_sessions_no_enforce_ordering(
+        self,
+        session,
+        db_parameters,
+    ) -> None:
+        try:
+            # create table
+            table_name = Utils.random_table_name()
+            session.create_dataframe(
+                [BASIC_TYPE_DATA1, BASIC_TYPE_DATA2]
+            ).write.save_as_table(table_name)
+
+            # create dataframe with enforce_ordering disabled
+            snow_dataframe = pd.read_snowflake(
+                f"(((SELECT * FROM {table_name})))", enforce_ordering=False
+            )
+
+            # create view
+            view_name = Utils.random_view_name()
+            assert (
+                "successfully created"
+                in pd.to_view(obj=snow_dataframe, name=view_name)[0]["status"]
+            )
+
+            # another session
+            new_session = Session.builder.configs(db_parameters).create()
+            pd.session = new_session
+
+            # accessing the created view in another session succeeds when enforce_ordering is disabled
+            res = new_session.sql(f"select * from {view_name}").collect()
+            assert len(res) == 2
+            new_session.close()
+
+        finally:
+            # cleanup
+            Utils.drop_view(session, view_name)
+            Utils.drop_table(session, table_name)
+            pd.session = session
+
+    @pytest.mark.parametrize("index", [True, False])
+    @pytest.mark.parametrize("index_labels", [None, ["my_index"]])
```

Comment:
prefer parametrizing over having logic in the test body

```suggestion
    @pytest.mark.parametrize(
        "index,index_labels,expected_index_columns",
        [
            (True, None, ["index"]),
            (True, ["my_index"], ["my_index"]),
            (False, None, []),
            (False, ["my_index"], []),
        ],
    )
```